### PR TITLE
[Hotfix] Add database filter to Server ExternalCatalog

### PR DESCRIPTION
**Pull request by zhongqishang**
## Why are the changes needed?

Add config table-filter for `listDatabases()`

## Brief change log

- Add filter 
- Remove deprecated config

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / **no**)
- If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by zhongqishang** [comment]
@HuangFru @wangtaohz @baiyangtx 

We have a large number of previous hive databases based on HMS, and 0.6 is not displayed on the page. I think we need to maintain this.

File: ams/server/src/main/java/com/netease/arctic/server/catalog/ExternalCatalog.java [hunk 1]
```diff
@@ -73,7 +73,14 @@ public boolean exist(String database, String tableName) {
...
   @Override
   public List<String> listDatabases() {
-    return doAs(() -> unifiedCatalog.listDatabases());
+    return doAs(
+        () ->
+            unifiedCatalog.listDatabases().stream()
+                .filter(
+                    database ->
+                        tableFilterPattern == null
+                            || tableFilterPattern.matcher(database + ".table_name").matches())
```
Comment (HuangFru) [inline ams/server/src/main/java/com/netease/arctic/server/catalog/ExternalCatalog.java:82]: What if the table filter is 'db.table' strict match?

Comment (HuangFru) [reply ams/server/src/main/java/com/netease/arctic/server/catalog/ExternalCatalog.java:82]: Maybe we do need to discuss whether the db filter parameter is still needed. As for me, too many Hive dbs indeed impact the experience.
cc @zhoujinsong @wangtaohz @baiyangtx .

**Review by baiyangtx — COMMENTED** [review]
The CI is failed. Could you please fit

**Comment by baiyangtx** [comment]
I agree with maintaining two filters, database-filter and table-filter. The table-filter specifically applies to db.table.

**Review by baiyangtx — APPROVED** [review]
LGTM, thanks for your contribution.

File: ams/api/src/main/java/com/netease/arctic/ams/api/properties/CatalogMetaProperties.java [hunk 2]
```diff
@@ -48,11 +48,7 @@ public class CatalogMetaProperties {
   // Deprecated from version v0.4.0, use KEY_WAREHOUSE
   @Deprecated public static final String KEY_WAREHOUSE_DIR = "warehouse.dir";
   public static final String KEY_WAREHOUSE = "warehouse";
-  /**
-   * @deprecated since 0.6.0, will be removed in 0.7.0; use {@link
-   *     CatalogMetaProperties#KEY_TABLE_FILTER} instead.
-   */
-  @Deprecated
+
```
Comment (zhoujinsong) [inline ams/api/src/main/java/com/netease/arctic/ams/api/properties/CatalogMetaProperties.java:51]: Should we change the property name to `database-filter`, to keep the same pattern with `table-filter`?

Comment (zhoujinsong) [reply ams/api/src/main/java/com/netease/arctic/ams/api/properties/CatalogMetaProperties.java:51]: BTW, we may add the new property into user docs.

Comment (zhongqishang) [reply ams/api/src/main/java/com/netease/arctic/ams/api/properties/CatalogMetaProperties.java:51]: Use `database.filter-regular-expression` just considering compatibility. 
Keeping the same style is more suitable for subsequent use, and `database.filter-regular-expression` has been deprecated in 0.6, so I agree to change it to `database-filter`.

Done.

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Pull request merged**